### DISCUSSION TITLE
TileLayer with ArcGIS Map Tile Service Map Appearing At Wrong Longitude...

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5575] [Q&A]
I am trying to use the [TileLayer ](https://deck.gl/docs/api-reference/geo-layers/tile-layer) with an [ArcGIS Map Tile Service](https://developers.arcgis.com/rest/services-reference/map-tile.htm).

I am running into an issue where the tiles render at the wrong latitude. When fully zoomed out the map renders in the top half of the deck canvas when I would expect it to be centered. With the map rendering in the wrong hemisphere the coordinates are all off... The spatial reference on the service is 4326; not sure if that has something to do with it. I created a [CodePen](https://codepen.io/jfuehner/pen/poNQYEo) showing the issue with the world/tiles rendering in the upper half of the canvas.

I had to specify a TileLayer `data` url in the format of z/y/x to get the tiles. The `z` also appears to need an offset of -1 to display the entire world at 0 zoom scale. I created a [CodePen ](https://codepen.io/jfuehner/pen/qBqQvEr) showing the issue with the entire world not rendering.

How can I make the map tiles render correctly?

```
const layer = new TileLayer({
  id: 'TileLayer',
  data: 'https://sampleserver1.arcgisonline.com/ArcGIS/rest/services/PublicSafety/PublicSafetyBasemap/MapServer/tile/{z}/{y}/{x}.png',
 
  renderSubLayers: props => {
    const {
      bbox: {west, south, east, north}
    } = props.tile;

    return new BitmapLayer(props, {
      data: null,
      image: props.data,
      bounds: [west, south, east, north]
    });
  }
});
```

**Comment by Pessimistress** [comment]
TileLayer in MapView (aka Web Mercator projection/EPSG:3857) only uses OSM tile encoding: https://deck.gl/docs/api-reference/geo-layers/tile-layer#indexing-system you need to use the OrthographicView if you want to render 4326.

See this thread
https://github.com/visgl/deck.gl/discussions/5539#discussioncomment-409221

**Reply by jfuehner** [reply]
@Pessimistress so switching to the OrthographicView causes the map to appear in the bottom right of the screen. See this [CodePen](https://codepen.io/jfuehner/pen/JjbeqvG). I have tried a number of different options but, clearly I am missing something with how to render a basemap correctly in this view and the sublayer (i.e., BitmapLayer)... 

Any help would be great!

**Reply by sclebal** [reply]
@jfuehner you have to update your initial view state to the format accepted by OrtographicView (you can see it [here](https://deck.gl/docs/api-reference/core/orthographic-view#view-state)). I've been experimenting with you CodePen and `target: [250, 120, 0],` seems a good value.

**Reply by jfuehner** [reply]
@Clebal thanks! That was super helpful. So, the `target` property is a `xyz` array?

It appears as though the `OrthographicView` `OrthographicController` does not support all the same capabilities the `MapController` supports... 

![image](https://user-images.githubusercontent.com/1843941/111547374-716bfa00-874f-11eb-81b6-0cc96de25e88.png)

The `OrthographicView` certainly does not appear to be a drop in replacement for the `MapView` even though I can render the 4326 projection. Definitely, would be nice if the `TileLayer` could support different projections or perhaps the `MapView` could support various projections in a more user friendly way.

**Reply by sclebal** [reply]
@jfuehner exactly, it's a xyz array.

**Comment by kylebarron** [comment]
As further commentary, I think the TileLayer _should_ allow the user to set the tiling parameters to render non-Web Mercator tiles in a geospatial view. I'd encourage you to read and comment on https://github.com/visgl/deck.gl/pull/5504. The goal there is to expose more tiling parameters as public exports and allow you to implement another tiling system such as 4326 yourself. See also https://github.com/visgl/deck.gl/discussions/5513#discussioncomment-407357
